RosNet/RosReader/RosBagReader.cs

```diff
@@ -12,6 +16,17 @@ namespace RosNet.RosReader
     /// </summary>
     public static class RosBagReader
     {
+        //Dictionary containing the names of all header fields in record
+        private static readonly Dictionary<OpCode, string[]> HeaderFieldsByOp = new Dictionary<OpCode, string[]>()
```

Comment:
Personally I would have implemented this as a static function on the OpCode `Enum`:
```csharp
static Set<string> Headers(OpCode code) => code switch 
{
    MessageData => new() {"conn", "time"},
    BagHeader => new() {"index_pos", "conn_count", "chunk_count"},
 // etc...
}
```